module/VuFind/src/VuFind/Http/CachingDownloader.php

```diff
@@ -117,17 +117,18 @@ protected function getDownloaderCache()
      *
      * @param string $cacheId             Cache ID
      * @param string $cacheOptionsSection Cache Options Section
+     * @param string $cacheOptionsFile    Config file defining the cache options
      *
      * @return void
      */
-    public function setUpCache(string $cacheId, string $cacheOptionsSection = null)
+    public function setUpCache(string $cacheId, string $cacheOptionsSection = null, string $cacheOptionsFile = null)
```

Comment:
How would you feel about something a little more like this?

```
    /**
     * Set up a different cache.
     *
     * @param string $cacheId             Cache ID
     * @param string $cacheOptionsSection Cache options config section suffix: if set, will be prepended to Config to
     * make up a section name in $cacheOptionsFile. If unset, indicates no special config settings if $cacheOptionsFile
     * is also null, or causes the default [Cache] section to be used if $cacheOptionsFile is not null.
     * @param string $cacheOptionsFile    Config file defining the cache options (defaults to config.ini)
     *
     * @return void
     */
    public function setUpCache(string $cacheId, string $cacheOptionsSection = null, string $cacheOptionsFile = null)
    {
        $this->cache = null;
        $this->cacheId = $cacheId;

        if ($cacheOptionsSection || $cacheOptionsFile) {
            $fullCacheOptionsSection = $cacheOptionsSection ? 'Cache_' . $cacheOptionsSection : 'Cache';
            $section = $this->configManager->get($cacheOptionsFile ?? 'config')->$fullCacheOptionsSection;
            $this->cacheOptions = !empty($section) ? $section->toArray() : [];
        }
    }
```

Maybe this is too unintuitive and convoluted to be worth changing, but it might be nice to just have [Cache] in ExternalVuFund.ini instead of the redundant [Cache_ExternalVuFind]. What do you think?

Reply:
I like getting rid of the redundancy, but I think that if we're re-using the structure of [Cache_*] from config.ini, we should be consistent and reuse the section name structure.  Also, since [Cache_ExternalVuFind] inherits its defaults from [Cache] in config.ini, it's slightly easier to follow that logic in the code if the sections are named differently.  And if we eventually allow for multiple ExternalVuFind caches, i.e. [Cache_ReShare] and others, and those two continue to inherit from config.ini [Cache], it will be much clearer if there is no [Cache] in ExternalVuFind.ini to confuse things.

Reply:
Good call, I renamed it [Cache_ExternalVuFind_Defaults]